src/hooks/useModal.tsx

```diff
@@ -0,0 +1,10 @@
+import { useState } from 'react';
+
+export const useModal = () => {
+  const [isShown, setIsShown] = useState<boolean>(false);
```

Comment:
Unnecessary typing here:
```suggestion
  const [isShown, setIsShown] = useState(false);
```
You only need to explicitly declare the type if the initial value is a null-value